enumerable_methods.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    while self[i]
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    my_each { |e| yield(index(e), e) }
+    self
+  end
```

Comment:
You seem to have swapped the order of the block variables here.  :point_right:  `each with index` :wink: 

`[2, 4, 6].each_with_index { |num, index| puts "#{num} -> #{index}" }` should output:

```
2 -> 0
4 -> 1
6 -> 2
```